pallets/claim/src/lib.rs

```diff
@@ -0,0 +1,222 @@
+		r.0.copy_from_slice(&raw);
+		Ok(r)
+	}
+}
+
+
+/// Configure the pallet by specifying the parameters and types on which it depends.
+pub trait Trait: frame_system::Trait {
+	/// Because this pallet emits events, it depends on the runtime's definition of an event.
+	type Event: From<Event<Self>> + Into<<Self as frame_system::Trait>::Event>;
+}
+
+// The pallet's runtime storage items.
+// https://substrate.dev/docs/en/knowledgebase/runtime/storage
+decl_storage! {
+    trait Store for Module<T: Trait> as ClaimModule {
+    	AdminId get(fn admin_id): T::AccountId;
+    	EndHeight get(fn end_height): u64;
+    	BurnedTransactions get(fn destroyed_transaction): map hasher(blake2_128_concat) EthereumTxHash => (EthereumAddress, u64);
+    	ClaimState get(fn claim_state): map hasher(blake2_128_concat) EthereumTxHash => bool;
+    }
+}
+
+// Pallets use events to inform users when important changes are made.
+// https://substrate.dev/docs/en/knowledgebase/runtime/events
+decl_event!(
+    pub enum Event<T> where AccountId = <T as frame_system::Trait>::AccountId {
+        /// Event emitted when a transaction has been stored.
+        ERC20TransactionStored(AccountId, EthereumTxHash, EthereumAddress, u64),
+        /// Event emitted when a transaction has been claimed.
+        ERC20TokenClaimed(AccountId, EthereumTxHash),
+        // ClaimDebug(AccountId, Vec<u8>, EthereumTxHash, EthereumAddress, EthereumAddress),
+    }
+);
+
+// Errors inform users that something went wrong.
+decl_error! {
+    pub enum Error for Module<T: Trait> {
+        /// Not administrator
+        NotAdministrator,
+        /// The transaction signature is invalid.
+        InvalidSignature,
+        /// The signer is not transaction sender.
+        NotTransactionSender,
+        /// The transaction hash doesn't exist
+        TxHashNotFound,
+        /// The transaction hash already exit
+        TxHashAlreadyExist,
+        /// The transaction has been claimed
+        TxAlreadyClaimed,
+        /// The transaction height less than end height
+        LessThanEndHeight,
+    }
+}
+
+// Dispatchable functions allows users to interact with the pallet and invoke state changes.
+// These functions materialize as "extrinsics", which are often compared to transactions.
+// Dispatchable functions must be annotated with a weight and must return a DispatchResult.
+decl_module! {
+	pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+      	// Errors must be initialized if they are used by the pallet.
+		type Error = Error<T>;
+      	fn deposit_event() = default;
+
+      	#[weight = 0]
+      	pub fn set_admin(origin, admin: T::AccountId) -> dispatch::DispatchResult {
+      		ensure_root(origin)?;
+      		Admin_Id::<T>::put(admin);
+            Ok(())
+        }
+
+
+        #[weight = 0]
+        pub fn store_erc20_burned_transaction(origin, height: u64, eth_tx_hash: EthereumTxHash, eth_address: EthereumAddress, amount:u64) -> dispatch::DispatchResult {
```

Comment:
It's better to push all the transactions within one block at a time. Otherwise we may end up in a situation where some transactions are not synced (e.g. the substrate transaction somehow failed, or the relayer script missed some transactions). By grouping all tx in a block together, we can make sure the sync is atomic at block level.